tests/purs/failing/FoldableInstance5.purs

```diff
@@ -1,9 +0,0 @@
--- @shouldFailWith CannotDeriveInvalidConstructorArg
-module FoldableInstance5 where
-
-import Prelude
-import Data.Foldable (class Foldable)
-import Data.Tuple (Tuple(..))
-
-data Test a = Test (Tuple a Int)
-derive instance Foldable Test
```

Comment:
Shouldn't this be moved to a passing test now?

Reply:
This and the other case are adequately covered (I felt) by [passing/DerivingFunctorFromBi.purs](https://github.com/purescript/purescript/pull/4420/files#diff-d952cbc4d102f894562bbba543ea5dcc61bd58513ddc29b7a11fa86df75e132eR10-R16). (The second case becomes `Test2 (Tuple (Array a) a)`, which is only slightly more sophisticated but exercises the same thing.)

Reply:
Ah, good point. I don't think I noticed the `Foldable`/`Traversable` instances there since the file name was specifically for `Functor`. However, using similar feedback as you suggested when I was working on just `Foldable`, shouldn't there be a test to verify the order of the fold works as expected? Or do you think the current `Foldable` tests are sufficient?

Reply:
I figured the current tests were sufficient—it's the same code paths being exercised in the `Foldable` and `Bifoldable` cases, as far as the order of the fold is concerned. But I could see an argument for adding those tests anyway, to ensure that if this code ever gets rewritten we don't regress behavior that would be part of the spec if we had one. Any leanings one way or the other?